academy/xl-cosmjs/first-steps.md

```diff
@@ -0,0 +1,637 @@
+
+Note that VSCode assists you to auto-complete [`StargateClient`](https://github.com/cosmos/cosmjs/blob/0f0c9d8a754cbf01e17acf51d3f2dbdeaae60757/packages/stargate/src/stargateclient.ts#L139) if you type <kbd>CTRL-Space</kbd> inside the `{}` of the `import` line.
+
+</HighlightBox>
+
+## Define your connection
+
+Next, you need to tell the client how to connect to the RPC port of your blockchain:
+
+```typescript
+const rpc = "https://rpc.sentry-01.theta-testnet.polypore.xyz"
+```
+
+Inside the `runAll` function you can initialize the connection and immediately check you connected to the right place:
+
+```typescript
+const runAll = async(): Promise<void> => {
+    const client = await StargateClient.connect(rpc)
+    console.log("chain id:", await client.getChainId(), ", height:", await client.getHeight())
+}
+```
+
+Run again to check with `npx yarn experiment`, and you get:
+
+```
+chain id: theta-testnet-001 , height: 9507032
+```
+
+## Get a balance
+
+Normally you would not yet have access to your user's address. However, for this exercise you need to know how many tokens Alice has, so add a temporary new command inside `runAll`: 
+
+```typescript
+console.log(
+    "Alice balances:",
+    await client.getAllBalances("cosmos17tvd4hcszq7lcxuwzrqkepuau9fye3dal606zf"),
+)
+```
+
+`getAllBalances` is used because the default token name is not yet known. When you run it again, you get:
+
+```
+Alice balances: []
+```
+
+If you just created this account, Alice's balance is zero. Alice will need tokens to be able to send transactions. A common practice with testnets is to expose faucets (services that send you test tokens for free, within limits).
+
+Cosmos' Theta has a dedicated [Discord channel](https://discord.com/channels/669268347736686612/953697793476821092/958291295741313024) where you can ask for tokens once per day _per Discord user_.
+
+Go to the faucet channel and get tokens for Alice:
+
+```
+$request cosmos17tvd4hcszq7lcxuwzrqkepuau9fye3dal606zf theta
+```
+
+The faucet bot replies with a transaction hash, alone or embedded in a block explorer URL:
+
+```
+✅ Hash ID: 540484BDD342702F196F84C2FD42D63FA77F74B26A8D7383FAA5AB46E4114A9B
+//Or
+✅  https://testnet.cosmos.bigdipper.live/transactions/540484BDD342702F196F84C2FD42D63FA77F74B26A8D7383FAA5AB46E4114A9B
+```
+
+Check that Alice received the tokens with `npx yarn experiment`, which should return:
+
+```
+Alice balances: [ { denom: 'uatom', amount: '10000000' } ]
+```
+
+That's 10 ATOM. After this confirmation you can comment out the balance query.
+
+## Get the faucet address
+
+As an exercise you want Alice to send some tokens back to the faucet, so you will need its address. You can request this from the faucet bot, but it is also possible to get it using the transaction hash.
+
+First you need to get the transaction:
+
+```typescript
+const faucetTx: IndexedTx = (await client.getTx(
+    "540484BDD342702F196F84C2FD42D63FA77F74B26A8D7383FAA5AB46E4114A9B",
+))!
+```
+
```

Comment:
I will add this separately to avoid bad formatting.

```suggestion

If your IDE does not do it for you, do not forget to import `IndexedTx` at the top of the file:

```typescript
import { IndexedTx } from "@cosmjs/stargate"
\`\`\`

```

> Also IndexedTx is not mentioned as an import in this tutorial yet so VScode throws an error here.

I can add that but I am generally reluctant to constantly add these details as I reckon that can turn condescending. Especially in this case where a right-click on `IndexedTx` offers the solution.